### ISSUE TITLE
[TC] Random bots are staying preparing to attack but doing nothing

### ISSUE THREAD

**Issue by ike3** [ike3/mangosbot#4]
Sometimes something prevents bot to reach target and they just stay at the distance doing nothing. Attacking the target or even getting close could trigger bot movement and attack. Seems more actual for spell casters.
Investigation is required.

**Comment by dmanbob** [comment]
Yes this issue seems to be most bots.  When I locate a random bot they are just standing there looking like they are ready to cast but do nothing. I have only seen one bot actually move in the 3 days I tried the new Trinity commits.

**Comment by Codex2000** [comment]
They target object's that only can be seen by GM but are not attackable so they stuck there forever trying to kill script triger objects.

**Comment by ike3** [comment]
![image](https://cloud.githubusercontent.com/assets/179914/4518178/be7e9c66-4c79-11e4-8f35-a7deae90761b.png)
I don't think so. Just created a java-based simple monitoring site which shows live bot states. Most bots have no target at all and they are just staying. The real issue is that the 'move random' strategy was missing. Added it in [ai-0.36-73] so bots should move properly now.

**Comment by Kittnz** [comment]
Basic? Looks nice :), i just can't wait for the bots to become more like normal players (ie doing stuff players should do)

**Comment by dmanbob** [comment]
yeah glad you found it makes sense I was trying to get bots to do move random and they didn't move until I used grind command

**Comment by cacodaemonic** [comment]
I have also noticed this happening if my voidwalker pet tags the 2nd of two mobs and gets the agro first, once the random bots finished off the first mob they completely ignore the second mob.  If I put my pet on passive I have yet to see this happen.

**Comment by bomba09** [comment]
Tested in depth,with melees bots. For example with a feral cat,if u don't move the mobs over him,he will attack only doing white melee attacks (so no yellow attacks with spells,and so low dps). he starts doing real damage only when mobs are over him and after waking uo up him with follow command.When there are a group of  mobs,it happens that he finish to kill one mob,an then he stays doing nothing even if there are other mobs near.

**Comment by FallenAngelX** [comment]
anyone having trouble w mages just jumping and not attacking??

**Comment by ike3** [comment]
I think the problem is in bool MovementAction::MoveTo(uint32 mapId, float x, float y, float z)

```
if (distance > sPlayerbotAIConfig.contactDistance)
```

So it the distance is less than 0.5f the bot is not moving. Without the check the bot would move indefinetely due to floating point calculation problems so the distance would never be zero.

Also there is another movement in bool MovementAction::MoveNear(WorldObject\* target, float distance)

```
distance += target->GetObjectSize() / 2.0f;
```

So the bot is supposed to be just near the target. It seems that for some mobs the size is incorrect and the bot is thinking that it is touching the target but the distance is not enough for melee/spell.

**Comment by bomba09** [comment]
I can only suggest I'm not a programmer :)
If u want i post my test on variuos pg's at level 80 vs heroic boss. I wrote my notes about every problematic class bot and my vote.
The fast one is that Death Knights,Rogues and Retribution paladins have serious problems because they doesn't attack/move or they do only white damages. The other classes have some issues but not so important. The best class i tested as melee,is the warrior fury,he tends to follow the tank and do the best to attack the mobs.
This is my table :+1: 
![cserrattura](https://cloud.githubusercontent.com/assets/9356444/5045351/db2ac24e-6bfb-11e4-89e1-6fa7f8e083b3.JPG)

**Comment by wrathix** [comment]
Wasn't sure where to send private messages, but just wanted to thank you for all the work you do with the bots.  Even at their current state, its a marvelous thing to be able to play WoW like this alone.

Please don't give up this great work!
TY!

**Comment by chrisxvin** [comment]
I still got this problem with version ai-0.36-144 on trinity-3.3.5.
I added a bot within my logged-in account, a priest, she didn't accept "heal" or "do heal" whisper. She tells "heal: impossible". But I tried to use this character, heal spell has no problem.
I set heal strategy with "co +heal", but she just give me an attack state, but do nothing there. No damage output, no heal output...

Random bots have same problem...
